### PR TITLE
feat(gatsby): make master build for Netlify

### PR DESCRIPTION
**Pull request by yjkimjunior**
just for test deployment

### CODE REVIEW

File: front/gatsby/src/pages/Validators.tsx [hunk 1]
```diff
@@ -4,35 +4,38 @@
...
-  const { data } = useQuery(LATEST_SESSION_QUERY, {
-    pollInterval: 10000,
-  });
+  
   const [sessionIndex, setSessionIndex] = useState<number>();
 
   useEffect(() => {
+    const { data } = useQuery(LATEST_SESSION_QUERY, {
+      pollInterval: 10000,
+    });
```
Comment (amaurymartiny) [inline front/gatsby/src/pages/Validators.tsx]: Nice! so this solves the apollo error on gatsby build? Do you have an idea why?

Comment (yjkimjunior) [reply front/gatsby/src/pages/Validators.tsx]: it's absolutley messed up. This makes it build, but it violates React rule of hooks in `gatsby develop`, i.e. hook cannot be called within a hook.

**Comment by yjkimjunior** [comment]
wow netlify doesn't allow the build command to be called "build" https://docs.netlify.com/configure-builds/troubleshooting-tips/#command-not-found

File: front/gatsby/gatsby-ssr.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// Copyright 2018-2020 @paritytech/Nomidot authors & contributors
```
Comment (amaurymartiny) [inline front/gatsby/gatsby-ssr.js]: I would do like that guy, put this in a file and import it from gatsby-{ssr,browser}

File: front/context/src/AccountsContext.tsx [hunk 3]
```diff
@@ -36,17 +36,23 @@ export function AccountsContextProvider(props: Props): React.ReactElement {
...
+      if (!extensions.length) {
+        throw new Error(
+          'No extension found. Please install PolkadotJS extension.'
+        );
+      }
+
+      setExtension(extensions[0]);
+      setIsReady(true);
+      setAccounts(await web3Accounts());
+    }
```
Comment (amaurymartiny) [inline front/context/src/AccountsContext.tsx]: could throw an error here too if no window

File: front/context/src/AccountsContext.tsx [hunk 4]
```diff
@@ -36,17 +36,23 @@ export function AccountsContextProvider(props: Props): React.ReactElement {
    * Fetch accounts from the extension
    */
   async function fetchAccounts(): Promise<void> {
-    const extensions = await web3Enable(originName);
-
-    if (!extensions.length) {
-      throw new Error(
-        'No extension found. Please install PolkadotJS extension.'
+    if (typeof window !== `undefined`) {
```
Comment (amaurymartiny) [inline front/context/src/AccountsContext.tsx]: there are multiple instances of this, we could put a constant `IS_SSR=window !== `undefined``

File: front/gatsby/src/components/Layout/Header/Header.tsx [hunk 5]
```diff
@@ -201,7 +201,7 @@ export function Header(): React.ReactElement {
         contentLeft={<h2>{APP_TITLE}</h2>}
         contentRight={
           <>
-            {accounts.length ? (
+            {accounts && accounts.length ? (
```
Comment (amaurymartiny) [inline front/gatsby/src/components/Layout/Header/Header.tsx]: is this `accounts &&` still needed?

File: front/gatsby/src/apollo.ts [hunk 6]
```diff
@@ -17,6 +18,7 @@ const wsLink = new WebSocketLink({
   options: {
     reconnect: true,
   },
+  webSocketImpl: process.env.NODE_ENV === 'production' ? ws : undefined,
```
Comment (amaurymartiny) [inline front/gatsby/src/apollo.ts]: Would add a comment here

**Comment by yjkimjunior** [comment]
acutally is there a Parity Netlify team I can do this deploy through?

File: front/gatsby/src/components/Layout/Header/Header.tsx (hunk 5 above)
Comment (yjkimjunior) [reply front/gatsby/src/components/Layout/Header/Header.tsx]: seems not

**Pull request merged**